apps/dashboard/app/javascript/packs/files/utils.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * Generates a filename that doesn't match that of an existing file in the current
+ * directory. Will append "_copy" if a file with the same name exists and append 
+ * `_${n}` where n is an integer increasing from 1 until a filename is found that 
+ * doesn't currently exist.
+ *
+ * @param      {string}  originalName   The original file name
+ * @returns    {string}  safeName       The file name edited to prevent overwriting
+ */
+export function dupSafeName(originalName) {
+  const currentFilenames = history.state.currentFilenames;
```

Comment:
Doesn't seem like you need `currentFilenames` here in this function.

Reply:
Oh no, I see it below. Hmnmm... seems like it shouldn't rely on state. Seems like it should be purely functional and not have to access `history`. 

Looking at the changes as a whole - I'm not sure what this is fixing. Or is it just a refactor of existing behavior?

Reply:
I see what's going on now. Yea we'd need to either need to pass in `currentFilenames` or just use history as is.  I guess I'm OK with just using history directly in this function.